client/errors.go

```diff
@@ -72,8 +101,7 @@ func (e volumeNotFoundError) Error() string {
 // IsErrVolumeNotFound returns true if the error is caused
 // when a volume is not found in the docker host.
 func IsErrVolumeNotFound(err error) bool {
-	_, ok := err.(volumeNotFoundError)
-	return ok
+	return IsErrNotFound(err)
```

Comment:
I think we're missing something - is `IsErrNotFound` is exported any of the volume|image|container error not found will pass this check but there's no way to tell if it was volume|image|container - I'm suggesting to have a filed (like `Op` filed in go net error if I'm not wrong) which tell us more about the error. Does this make sense? /cc @stevvooe